sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/FlatteningSerializer.java

```diff
@@ -291,9 +309,25 @@ public void resolve(SerializerProvider provider) throws JsonMappingException {
         }
     }
 
-    @Override
-    public void serializeWithType(Object value, JsonGenerator gen, SerializerProvider provider,
-        TypeSerializer typeSerializer) throws IOException {
-        serialize(value, gen, provider);
+    private void propertyOnlyFlattenSerialize(Object value, JsonGenerator jgen, ObjectNode objNode) throws IOException {
+        for (BeanPropertyDefinition beanPropertyDefinition : beanDescription.findProperties()) {
+            if (jsonPropertiesWithJsonFlatten.contains(beanPropertyDefinition.getName())) {
+                String[] splitNames = SPLIT_FLATTEN_PROPERTY_PATTERN.split(beanPropertyDefinition.getName());
```

Comment:
It seems none of the above 3 lines are not specific, but done on each call, could something be pre-cached/pre-calculated for optimization reasons?

Reply:
I believe it would be possible to cache the properties and their split names if needed